operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -59,30 +59,61 @@ func (c *Controller) initializeQueue() {
 		logfields.WorkQueueBurstLimit:  c.rateLimit.current.Burst,
 		logfields.WorkQueueSyncBackOff: defaultSyncBackOff,
 	}).Info("CES controller workqueue configuration")
-	c.queue = workqueue.NewRateLimitingQueueWithConfig(
-		workqueue.NewItemExponentialFailureRateLimiter(defaultSyncBackOff, maxSyncBackOff),
+	c.rateLimiter = workqueue.NewItemExponentialFailureRateLimiter(defaultSyncBackOff, maxSyncBackOff)
```

Comment:
add a small comment explaining why we can have only one rateLimiter as the keys should not be present in both queues